File: Bars.GkhGJI/Entities/ProtocolRSO/ProtocolRSO.cs
namespace Bars.GkhGji.Entities
{
    using System;

    using Gkh.Entities;
    using Enums;

    /// <summary>
    /// Протокол прокуратуры
    /// </summary>
    public class ProtocolRSO : DocumentGji
    {
        /// <summary>
        /// Тип исполнителя документа
        /// </summary>
        public virtual ExecutantDocGji Executant { get; set; }

        /// <summary>
        /// Тип РСО составившей протокол
        /// </summary>
        public virtual TypeSupplierProtocol TypeSupplierProtocol { get; set; }

        /// <summary>
        /// Контрагент
        /// </summary>
        public virtual Contragent Contragent { get; set; }

        /// <summary>
        /// Физическое лицо
        /// </summary>
        public virtual string PhysicalPerson { get; set; }

        /// <summary>
        /// Реквизиты физ. лица
        /// </summary>
        public virtual string PhysicalPersonInfo { get; set; }

        /// <summary>
        /// Газоснабжающая организация, вынесшая протокол
        /// </summary>
        public virtual Contragent GasSupplier { get; set; }

        /// <summary>
        /// Дата поступления в ГЖИ
        /// </summary>
        public virtual DateTime? DateSupply { get; set; }
    }
}

File: Bars.GkhGJI/libs/B4/model/ProtocolRSO.js
Ext.define('B4.model.ProtocolRSO', {
    extend: 'B4.model.DocumentGji',
    idProperty: 'Id',
    proxy: {
        type: 'b4proxy',
        controllerName: 'ProtocolRSO'
    },
    fields: [
        { name: 'Inspection', defaultValue: null },
        { name: 'InspectionId' },
        { name: 'Executant', defaultValue: null },
        { name: 'Contragent', defaultValue: null },
        { name: 'GasSupplier', defaultValue: null },
        { name: 'State', defaultValue: null },
        { name: 'DateSupply' },
        { name: 'PhysicalPerson' },
        { name: 'DocUIN' },
        { name: 'PhysicalPersonInfo' },
        { name: 'TypeDocumentGji', defaultValue: 190 },
        { name: 'TypeSupplierProtocol', defaultValue: 0 },
        { name: 'Koap' },
    ]
});